src/test/java/com/radixdlt/serialization/mapper/RadixObjectMapperConfiguratorTest.java

```diff
@@ -0,0 +1,88 @@
+package com.radixdlt.serialization.mapper;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import com.radixdlt.serialization.ClassScanningSerializationPolicy;
+import com.radixdlt.serialization.DsonOutput;
+import com.radixdlt.serialization.DsonOutput.Output;
+import com.radixdlt.serialization.Serialization;
+import com.radixdlt.serialization.SerializerConstants;
+import com.radixdlt.serialization.SerializerDummy;
+import com.radixdlt.serialization.SerializerId2;
+import com.radixdlt.serialization.core.ClasspathScanningSerializerIds;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class RadixObjectMapperConfiguratorTest {
+	private final Serialization serialization = Serialization.create(
+			ClasspathScanningSerializerIds.create(),
+			new ClassScanningSerializationPolicy(ImmutableList.of(OrderedPropertyBean.class)) {
+			}
+	);
+
+	@Test
+	public void propertyOrderingIsCorrect() {
+		var inputBean =
+				new OrderedPropertyBean("b", "z", "a", "x").propertyC("c");
+
+		var serialized = serialization.toDson(inputBean, Output.ALL);
+
+		assertEquals(">aaaaababacacjserializervtest.property_orderingaxaxazaz~", toText(serialized));
+	}
+
+	//Somewhat artificial transformation to shift output into printable range
+	private static String toText(byte[] input) {
+		var output = new char[input.length];
+
+		for (int i = 0; i < input.length; i++) {
+			output[i] = input[i] > 0 ? (char) input[i] : (char) (127 + input[i]);
+		}
+		return new String(output);
+	}
+
+	@SerializerId2("test.property_ordering")
+	public static class OrderedPropertyBean {

```

Comment:
I would also like to see a test here to make sure that fields were ordered across all superclasses too.

One way to do this would be to move fields a, b, c into an abstract superclass.

Reply:
I think it'll be better to add separate test for this.